tensornetwork/backends/tensorflow/tensordot2.py

```diff
@@ -99,37 +120,34 @@ def _tensordot_reshape(a, axes, is_right_term=False):
     """
     if a.get_shape().is_fully_defined() and isinstance(axes, (list, tuple)):
       shape_a = a.get_shape().as_list()
+      # NOTE: This will fail if axes contains any tensors
```

Comment:
I don't think we should ever have axes be tensors, but if you add typing we can make sure that never happens ;)